WordPress/gutenberg
# Remove trash button in post/page inspector

Part of https://github.com/WordPress/gutenberg/issues/59689.

## What?
Remove the 'Move to trash' button beneath post meta.

## Why?
1. It's overly prominent.
2. Inconsistent with other trashing flows (e.g. templates and patterns).
3. Duplicative with the new actions menu.

| Before | After |
| --- | --- |
| <img width="280" alt="Screenshot 2024-05-20 at 14 13 34" src="https://github.com/WordPress/gutenberg/assets/846565/a914668f-9476-4a3b-adda-56cbe9da66d4"> |  <img width="278" alt="Screenshot 2024-05-20 at 14 13 01" src="https://github.com/WordPress/gutenberg/assets/846565/9704048e-0969-4637-97c1-b2ef40fb24c6"> |


## Testing Instructions
1. Open a post or page.
2. Open the Inspector and select the Post/Page tab.
3. Check the 'Move to trash' button is no longer visible.
4. Check the 'Move to trash' action in the ellipsis menu works as expected.
5. Check other post types (pattern, template, navigation menu).
